frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -0,0 +1,385 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::helper;
+use syn::spanned::Spanned;
+use quote::ToTokens;
+
+/// List of additional token to be used for parsing.
+mod keyword {
+	syn::custom_keyword!(Config);
+	syn::custom_keyword!(From);
+	syn::custom_keyword!(T);
+	syn::custom_keyword!(I);
+	syn::custom_keyword!(Get);
+	syn::custom_keyword!(config);
+	syn::custom_keyword!(IsType);
+	syn::custom_keyword!(Event);
+	syn::custom_keyword!(constant);
+	syn::custom_keyword!(frame_system);
+	syn::custom_keyword!(disable_frame_system_supertrait_check);
+}
+
+/// Input definition for the pallet config.
+pub struct ConfigDef {
+	/// The index of item in pallet module.
+	pub index: usize,
+	/// Wheither the trait has instance (i.e. define with `Config<I = ()>`)
+	pub has_instance: bool,
+	/// Const associated type.
+	pub consts_metadata: Vec<ConstMetadataDef>,
+	/// Wether the trait has the associated type `Event`, note that those bounds are checked:
```

Comment:
"whether" on these two. maybe best with a find and replace

Reply:
damn I wasn't aware my spelling of whether was so inconsistent, I'll do check them, and try to run spellcheck on the codebase

Reply:
they should be all fixed now